encoding/unstable_base64.ts

```diff
@@ -0,0 +1,241 @@
+ *
+ * encodeRawBase64(output, i, o, "Base64");
+ * assertEquals(
+ *   new TextDecoder().decode(output),
+ *   "data:url/fake," + encodeBase64(await Deno.readFile("./deno.lock"), "Base64"),
+ * );
+ * ```
+ */
+export function encodeRawBase64(
+  buffer: Uint8Array_,
+  i: number,
+  o: number,
+  format: Base64Format,
+): number {
+  const max = calcMax(buffer.length - i);
+  if (max > buffer.length - o) throw new RangeError("Buffer too small");
+  o = encode(buffer, i, o, alphabet[format], padding);
+  if (format === "Base64Url") {
+    i = buffer.indexOf(padding, o - 2);
+    if (i > 0) return i;
+  }
+  return o;
+}
+
+/**
+ * `decodeBase64` takes an input source and decodes it into a
+ * {@linkcode Uint8Array<ArrayBuffer>} using the specified format.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @param input The input source to decode.
+ * @param format The format to use for decoding.
+ * @returns The decoded {@linkcode Uint8Array<ArrayBuffer>}.
+ *
+ * @example Basic Usage
+ * ```ts
+ * import { assertEquals } from "@std/assert";
+ * import { decodeBase64 } from "@std/encoding/unstable-base64";
+ *
+ * assertEquals(
+ *   decodeBase64("SGVsbG8gV29ybGQ=", "Base64"),
+ *   new TextEncoder().encode("Hello World"),
+ * );
+ *
+ * assertEquals(
+ *   decodeBase64("SGVsbG8gV29ybGQ", "Base64Url"),
+ *   new TextEncoder().encode("Hello World"),
+ * );
+ * ```
+ */
+export function decodeBase64(input: string, format: Base64Format): Uint8Array_ {
+  const output = new TextEncoder().encode(input) as Uint8Array_;
+  return output
+    .subarray(
+      0,
+      decode(output, 0, 0, rAlphabet[format], padding),
+    );
+}
+
+/**
+ * `decodeRawHex` is a low-level function that decodes a
+ * {@linkcode Uint8Array<ArrayBuffer>} from hexadecimal in place. Param
+ * {@linkcode i} must be greater than or equal to param {@linkcode o}. The
+ * function assumes that the encoded data starts at param {@linkcode i} and ends
+ * at the end of the buffer.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @param buffer The buffer to decode in place.
+ * @param i The index of where the encoded data starts reading from.
+ * @param o The index of where the decoded data starts writing to.
+ * @param format The format to use for decoding.
+ * @returns The index of where the decoded data finished writing to.
+ *
+ * @example Basic Usage
+ * ```ts
+ * import { assertEquals } from "@std/assert";
+ * import {
+ *   decodeRawBase64,
+ *   encodeBase64,
+ *   type Uint8Array_,
+ * } from "@std/encoding/unstable-base64";
+ *
+ * let buffer = new TextEncoder().encode(
+ *   "data:url/fake," + encodeBase64(await Deno.readFile("./deno.lock"), "Base64"),
+ * ) as Uint8Array_;
+ *
+ * const i = buffer.indexOf(",".charCodeAt(0)) + 1;
+ * const o = decodeRawBase64(buffer, i, i, "Base64");
+ *
+ * buffer = buffer.subarray(i, o);
+ * assertEquals(buffer, await Deno.readFile("./deno.lock"));
+ * ```
+ */
+export function decodeRawBase64(
```

Comment:
When do you think the users would prefer this API over `decodeBase64`?

Reply:
The only real use case for this API and the `encodeRawBase64` is if you're input and output is a `Uint8Array`. Say you're creating a file and part of it needs to be base64 for whatever consumes it, this API lets you skip the conversion to a string and back and in the same instance use less memory. 

Outside of where the encoding is embedded in other information, I doubt it would find much use.